packages/nextjs/components/scaffold-eth/Contract/utilsContract.tsx

```diff
@@ -18,14 +18,21 @@ const getFunctionInputKey = (functionName: string, input: AbiParameter, inputInd
  * @param e - error object
  * @returns {string} parsed error string
  */
-const getParsedError = (e: any | BaseViemError): string => {
-  let message = e.message ?? "An unknown error occurred";
-
+const getParsedError = (e: any): string => {
+  let message: string = e.message ?? "An unknown error occurred";
   if (e instanceof BaseViemError) {
     if (e.details) {
       message = e.details;
     } else if (e.shortMessage) {
       message = e.shortMessage;
+      const cause = e.cause as any as { data?: DecodeErrorResultReturnType };
```

Comment:
Had to do this :( , I think there is some problem with `viem's`  [`BaseError`](https://viem.sh/docs/glossary/errors.html#errors) type although on runtime it `e.cause.data` is present when function always revert but `viem` type doesn't have typing for it and is typed as `unknown`. 


I also tried an approach by narrowing the type with `ContractFunctionExecutionError` instead of `BaseViemError` but it too does not have proper typing for `e.cause.data`

The reason we are trying to get `e.cause.data` because it is the only place from where you can get details about Custom error checkout this : https://github.com/wevm/viem/discussions/589#discussioncomment-6028456

Reply:
Since `e.cause` is of type `unknown`, you should be able to cast it directly to the desired type without the `as any` in the middle.

So this should work:
```ts
const cause = e.cause as { data?: DecodeErrorResultReturnType };
```